src/core_plugins/elasticsearch/lib/create_proxy.js

```diff
@@ -3,48 +3,57 @@ import mapUri from './map_uri';
 import { resolve } from 'url';
 import { assign } from 'lodash';
 
-function createProxy(server, method, route, config) {
-
-  const options = {
-    method: method,
-    path: createProxy.createPath(route),
-    config: {
-      timeout: {
-        socket: server.config().get('elasticsearch.requestTimeout')
-      }
-    },
-    handler: {
-      proxy: {
-        mapUri: mapUri(server),
-        agent: createAgent(server),
-        xforward: true,
-        timeout: server.config().get('elasticsearch.requestTimeout'),
-        onResponse: function (err, responseFromUpstream, request, reply) {
-          if (err) {
-            reply(err);
-            return;
-          }
-
-          if (responseFromUpstream.headers.location) {
-            // TODO: Workaround for #8705 until hapi has been updated to >= 15.0.0
-            responseFromUpstream.headers.location = encodeURI(responseFromUpstream.headers.location);
-          }
-
-          reply(null, responseFromUpstream);
-        }
-      }
-    },
+function createProxy(server, method, path, config) {
+  const proxies = new Map([
+    ['/elasticsearch', server.plugins.elasticsearch.getCluster('data')],
+    ['/es_admin', server.plugins.elasticsearch.getCluster('admin')]
+  ]);
+
+  const responseHandler = function (err, upstreamResponse, request, reply) {
+    if (err) {
+      reply(err);
+      return;
+    }
+
+    if (upstreamResponse.headers.location) {
+      // TODO: Workaround for #8705 until hapi has been updated to >= 15.0.0
+      upstreamResponse.headers.location = encodeURI(upstreamResponse.headers.location);
+    }
+
+    reply(null, upstreamResponse);
   };
 
-  assign(options.config, config);
+  for (let [proxyPrefix, cluster] of proxies) {
```

Comment:
I'm pretty sure this should be const since it's scoped to the loop block rather than the parent function.